File: SDKs/twilio.com/twilio_voice_v1/1.36.0/python/src/sdk/models/operations/listconnectionpolicytarget.py
from dataclasses import dataclass, field
from typing import List,Optional
from dataclasses_json import dataclass_json
from sdk.models import shared
LIST_CONNECTION_POLICY_TARGET_SERVERS = [
	"https://voice.twilio.com",
]


@dataclass
class ListConnectionPolicyTargetPathParams:
    connection_policy_sid: str = field(default=None, metadata={'path_param': { 'field_name': 'ConnectionPolicySid', 'style': 'simple', 'explode': False }})
    

@dataclass
class ListConnectionPolicyTargetQueryParams:
    page_size: Optional[int] = field(default=None, metadata={'query_param': { 'field_name': 'PageSize', 'style': 'form', 'explode': True }})
    

@dataclass
class ListConnectionPolicyTargetSecurity:
    account_sid_auth_token: shared.SchemeAccountSidAuthToken = field(default=None, metadata={'security': { 'scheme': True, 'type': 'http', 'sub_type': 'basic' }})
    

@dataclass
class ListConnectionPolicyTargetRequest:
    server_url: Optional[str] = field(default=None)
    path_params: ListConnectionPolicyTargetPathParams = field(default=None)
    query_params: ListConnectionPolicyTargetQueryParams = field(default=None)
    security: ListConnectionPolicyTargetSecurity = field(default=None)
    

@dataclass_json
@dataclass
class ListConnectionPolicyTarget200ApplicationJSONMeta:
    first_page_url: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'first_page_url' }})
    key: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'key' }})
    next_page_url: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'next_page_url' }})
    page: Optional[int] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'page' }})
    page_size: Optional[int] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'page_size' }})
    previous_page_url: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'previous_page_url' }})
    url: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'url' }})
    

@dataclass_json
@dataclass
class ListConnectionPolicyTarget200ApplicationJSONListConnectionPolicyTargetResponse:
    meta: Optional[ListConnectionPolicyTarget200ApplicationJSONMeta] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'meta' }})
    targets: Optional[List[shared.VoiceV1ConnectionPolicyConnectionPolicyTarget]] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'targets' }})
    

@dataclass
class ListConnectionPolicyTargetResponse:
    content_type: str = field(default=None)
    list_connection_policy_target_response: Optional[ListConnectionPolicyTarget200ApplicationJSONListConnectionPolicyTargetResponse] = field(default=None)
    status_code: int = field(default=None)
    

File: SDKs/twilio.com/twilio_voice_v1/1.36.0/typescript/src/sdk/models/operations/listconnectionpolicytarget.ts
import { Metadata, SpeakeasyBase } from "../../../internal/utils/utils";
import * as shared from "../shared";
export const LISTCONNECTIONPOLICYTARGET_SERVERS = [
	"https://voice.twilio.com",
];



export class ListConnectionPolicyTargetPathParams extends SpeakeasyBase {
  @Metadata({ data: "pathParam, style=simple;explode=false;name=ConnectionPolicySid" })
  connectionPolicySid: string;
}


export class ListConnectionPolicyTargetQueryParams extends SpeakeasyBase {
  @Metadata({ data: "queryParam, style=form;explode=true;name=PageSize" })
  pageSize?: number;
}


export class ListConnectionPolicyTargetSecurity extends SpeakeasyBase {
  @Metadata({ data: "security, scheme=true;type=http;subtype=basic" })
  accountSidAuthToken: shared.SchemeAccountSidAuthToken;
}


export class ListConnectionPolicyTargetRequest extends SpeakeasyBase {
  @Metadata()
  serverUrl?: string;

  @Metadata()
  pathParams: ListConnectionPolicyTargetPathParams;

  @Metadata()
  queryParams: ListConnectionPolicyTargetQueryParams;

  @Metadata()
  security: ListConnectionPolicyTargetSecurity;
}


export class ListConnectionPolicyTargetListConnectionPolicyTargetResponseMeta extends SpeakeasyBase {
  @Metadata({ data: "json, name=first_page_url" })
  firstPageUrl?: string;

  @Metadata({ data: "json, name=key" })
  key?: string;

  @Metadata({ data: "json, name=next_page_url" })
  nextPageUrl?: string;

  @Metadata({ data: "json, name=page" })
  page?: number;

  @Metadata({ data: "json, name=page_size" })
  pageSize?: number;

  @Metadata({ data: "json, name=previous_page_url" })
  previousPageUrl?: string;

  @Metadata({ data: "json, name=url" })
  url?: string;
}


export class ListConnectionPolicyTargetListConnectionPolicyTargetResponse extends SpeakeasyBase {
  @Metadata({ data: "json, name=meta" })
  meta?: ListConnectionPolicyTargetListConnectionPolicyTargetResponseMeta;

  @Metadata({ data: "json, name=targets", elemType: shared.VoiceV1ConnectionPolicyConnectionPolicyTarget })
  targets?: shared.VoiceV1ConnectionPolicyConnectionPolicyTarget[];
}


export class ListConnectionPolicyTargetResponse extends SpeakeasyBase {
  @Metadata()
  contentType: string;

  @Metadata()
  listConnectionPolicyTargetResponse?: ListConnectionPolicyTargetListConnectionPolicyTargetResponse;

  @Metadata()
  statusCode: number;
}
